src/Dapr.Jobs/DaprJobsGrpcClient.cs

```diff
@@ -0,0 +1,312 @@
+
+        var envelope = new Autogenerated.GetJobRequest { Name = jobName };
+
+        var callOptions = CreateCallOptions(headers: null, cancellationToken);
+        Autogenerated.GetJobResponse response;
+
+        try
+        {
+            response = await client.GetJobAlpha1Async(envelope, callOptions);
+        }
+        catch (RpcException ex)
+        {
+            throw new DaprException(
+                "Get job operation failed: the Dapr endpoint indicated a failure. See InnerException for details.", ex);
+        }
+        
+        return new JobDetails
+        {
+            DueTime = response.Job.DueTime is not null ? DateTime.Parse(response.Job.DueTime) : null,
+            TTL = response.Job.Ttl is not null ? DateTime.Parse(response.Job.Ttl) : null,
+            RepeatCount = response.Job.Repeats == default ? null : response.Job.Repeats,
+            Payload = response.Job.Data.ToByteArray()
+        };
+    }
+
+    /// <summary>
+    /// Deletes the specified job.
+    /// </summary>
+    /// <param name="jobName">The name of the job.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    /// <returns></returns>
+    [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+    public override async Task DeleteJobAsync(string jobName, CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrWhiteSpace(jobName))
+            throw new ArgumentNullException(nameof(jobName));
+
+        var envelope = new Autogenerated.DeleteJobRequest { Name = jobName };
+
+        var callOptions = CreateCallOptions(headers: null, cancellationToken);
+
+        try
+        {
+            await client.DeleteJobAlpha1Async(envelope, callOptions);
+        }
+        catch (RpcException ex)
+        {
+            throw new DaprException(
+                "Delete job operation failed: the Dapr endpoint indicated a failure. See InnerException for details.",
+                ex);
+        }
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            this.channel.Dispose();
+            this.httpClient.Dispose();
+        }
+    }
+
+    private CallOptions CreateCallOptions(Metadata? headers, CancellationToken cancellationToken)
+    {
+        var callOptions = new CallOptions(headers: headers ?? new Metadata(), cancellationToken: cancellationToken);
+
+        callOptions.Headers!.Add("User-Agent", UserAgent().ToString());
+
+        if (apiTokenHeader is not null)
+            callOptions.Headers.Add(apiTokenHeader.Value.Key, apiTokenHeader.Value.Value);
+
+        return callOptions;
+    }
+
+    /// <summary>
+    /// Returns the value for the User-Agent.
+    /// </summary>
+    /// <returns>A <see cref="ProductInfoHeaderValue"/> containing the value to use for the User-Agent.</returns>
+    private static ProductInfoHeaderValue UserAgent()
```

Comment:
Could this be generated just once and reused rather than on every call?

Reply:
Sure thing